### PR TITLE
Elyra Pipelines Lack Proper Task Names

### PR DESCRIPTION
**Pull request by pnaik1**
Closes: #1507 

## Description
#1507 
The elyra pipeline was lacking the proper task name in data science pipeline 
you can check the changes in pipelines tab :
Screenshot from 2023-07-27 15-23-10
and also in the triggered run tab
Screenshot from 2023-07-27 13-00-04

## How Has This Been Tested?
1) Create a workbench with notebook image that contain elyra extension (Standard data science, TensorFlow, TrustyAI, or PyTorch  ) , In the cluster, that are insecured , above env vars are needed to be set in workbenches (PIPELINES_SSL_SA_CERTS=/var/run/secrets/kubernetes.io/serviceaccount/ca.crt)
2) You have created and configured a pipeline server within the data science project that contains your workbench.
3)You have access to S3-compatible storage
4)Launch an Elyra Notebook
5)Build any pipeline of any number of tasks
6)Launch DS Pipelines Dashboard UI and view the Pipeline/Run views

## Test Impact
no test coverage

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit tests & storybook for related changes)

If you have UI changes: 

- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change.

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

File: frontend/src/concepts/pipelines/content/pipelinesDetails/pipeline/SelectedTaskDrawerContent.tsx [hunk 1]
```diff
@@ -28,7 +28,8 @@ const SelectedTaskDrawerContent: React.FC<SelectedTaskDrawerContentProps> = ({ t
     >
       <DrawerHead>
         <Title headingLevel="h2" size="xl">
-          {task.name}
+          {task.taskSpec?.metadata?.annotations['pipelines.kubeflow.org/task_display_name'] ||
```
Comment (lucferbux) [inline frontend/src/concepts/pipelines/content/pipelinesDetails/pipeline/SelectedTaskDrawerContent.tsx:31]: checking `PipelineRunTask` taskSpec is not optional (nor metadata inside `PipelineRunTaskSpec`) so either marked them as optional if you are sure they are not mandatory or remove the optional chaining. We should be mindful of our types.

**Review by lucferbux — APPROVED** [review]
/lgtm

**Comment by pnaik1** [comment]
@lucferbux , ahh  I got a catch in other pipeline , 
Screenshot from 2023-07-27 21-04-27
this was because some nodes in the pipeline don't have metadata and annotation 
so may be I think we need to keep this variable optional? what do you suggest?

**Comment by lucferbux** [comment]
> @lucferbux , ahh I got a catch in other pipeline , Screenshot from 2023-07-27 21-04-27 this was because some nodes in the pipeline don't have metadata and annotation so may be I think we need to keep this variable optional? what do you suggest?

Sorry, my bad, I thought you checked the types.

No, you need to be mindful of the types. You SHOULD then mark annotations as optional in the `PipelineRunStaskSpec` and then perform the optional chaining in the object.

My comment above was not against using optional chaining to access, but to use it correctly with typescript, we use types to ensure type safety, so in theory we shouldn't optional access an attribute we know is mandatory, in this kind of scenarios the golden rule is update the type and then wrap our code around it.

**Comment by pnaik1** [comment]
> > @lucferbux , ahh I got a catch in other pipeline , Screenshot from 2023-07-27 21-04-27 this was because some nodes in the pipeline don't have metadata and annotation so may be I think we need to keep this variable optional? what do you suggest?
> 
> Sorry, my bad, I thought you checked the types.
> 
> No, you need to be mindful of the types. You SHOULD then mark annotations as optional in the `PipelineRunStaskSpec` and then perform the optional chaining in the object.
> 
> My comment above was not against using optional chaining to access, but to use it correctly with typescript, we use types to ensure type safety, so in theory we shouldn't optional access an attribute we know is mandatory, in this kind of scenarios the golden rule is update the type and then wrap our code around it.

Yeah , I had double checked with 2 different pipelines , It was working perfectly fine ...I thought giving it mandatory will work in all scenario, but nope  sometime we have pipelines with no metadata and annotations , there it will cause an error

**Comment by lucferbux** [comment]
> > > @lucferbux , ahh I got a catch in other pipeline , Screenshot from 2023-07-27 21-04-27 this was because some nodes in the pipeline don't have metadata and annotation so may be I think we need to keep this variable optional? what do you suggest?
> > 
> > 
> > Sorry, my bad, I thought you checked the types.
> > No, you need to be mindful of the types. You SHOULD then mark annotations as optional in the `PipelineRunStaskSpec` and then perform the optional chaining in the object.
> > My comment above was not against using optional chaining to access, but to use it correctly with typescript, we use types to ensure type safety, so in theory we shouldn't optional access an attribute we know is mandatory, in this kind of scenarios the golden rule is update the type and then wrap our code around it.
> 
> my bad I had double checked with 2 different pipelines , It was working perfectly fine ...I thought giving it mandatory will work in all scenario, but nope sometime we have pipelines with no metadata and annotations , there it will cause an error

No worries!

Yeah, we need to be extra careful, sometimes we log types without double checking the attributes so be aware there might be issues, we need to double down our efforts on tracking the right types.

Thanks again for all the effort!

**Comment by pnaik1** [comment]
@lucferbux I have made the changes , and I completely understand the importance of being cautious , Thank you

**Review by lucferbux — APPROVED** [review]
/lgtm

**Comment by lucferbux** [comment]
> @lucferbux I have made the changes , and I completely understand the importance of being cautious , Thank you

Thanks a lot 🎉

**Review by manaswinidas — APPROVED** [review]
/lgtm

I can see the task names
Screenshot 2023-07-28 at 5 06 51 PM
Screenshot 2023-07-28 at 5 07 05 PM

**Review by Gkrumbach07 — APPROVED** [review]
/lgtm

this can be merged

**Pull request merged**